### PR TITLE
fastlane: Better changelogs

### LINKED ISSUE #2822: Only list merge commits in automated nightly changelogs
I really just want to see what PRs were merged, and doing this would remove a lot of the noise from the message.

### PR DESCRIPTION
**Pull request by rye**
This PR resolves #2822 by improving the iOS changelog format.

We now go from something like this:
```
*   e399b3ec Merge pull request #3045 from StoDevX/greenkeeper/react-native-0.57.0
|\  
| * 64d82112 remove fix-rn script
| * 79e66da2 disable [scrollEnabled] prop since it crashes RN@0.57.5
| *   40441dab Merge pull request #3185 from StoDevX/rn-0.57.5--remove-hsps
| |\  
| | * 5e916782 ios: Add HEADER_SEARCH_PATHS back to OneSignal targets
| | * 04d1da4f ios: Add back react-native-onesignal to pbxproj
| | * f0a42fd5 iOS: Try completely removing HEADER_SEARCH_PATHS
| * | 216d40d7 restore the accidentally-removed onesignal typedef
| * | c0003614 bump detox to v9.1.2
| * |   55a8ec62 Merge pull request #3184 from StoDevX/rn-0.57.5-remove-rc-from-hsp
| |\ \  
| | |/  
| | * 218edfd0 ios: Also remove ReactCommon from Release HSPs
| | * 9778d0c0 iOS: Try removing ReactCommon from the header search paths
| * | f4486778 remove react-navigation flow exclusion
| * | 7117a35e remove react-theme-provider flow exclusion
| * | ace369b7 update rn-linear-gradient to 2.4.4
| * | d19aef5e update callstack/react-theme-provider to 1.0.7
| * | 5872e53c RN 0.57.2 added [scrollEnabled] to the flow types
| * |   2e109eeb Merge pull request #3183 from StoDevX/rn-0.57.5--master-sync
| |\ \  
| | |/  
| |/|   
| | * f49bfd2b modules/lists/list-row: Remove unused eslint-disable directive
| | *   aea005fd Merge branch 'master' into greenkeeper/react-native-0.57.0
| | |\  
| |/ /  
| * | cf350ed1 yarn: Deduplicate the lockfile
| * | db893aa7 start update to v0.57.5
| * | 48978d03 bump eslint targeted flow version to 0.78
| * | 750d4c7c bump eslint target react version to 16.6
| * | 34e7a419 yarn: Upgrade react-native to 0.57.4
| * |   b672ab3f Merge branch 'master' into greenkeeper/react-native-0.57.0
| |\ \  
[...]
```
to something much more manageable and understandable like this:
```
Merged Pull Requests as of v2.6.3-470-ge399b3ec:

e399b3ec Merge pull request #3045 from StoDevX/greenkeeper/react-native-0.57.0
161270ad Merge pull request #3189 from StoDevX/express-bus-update
40441dab Merge pull request #3185 from StoDevX/rn-0.57.5--remove-hsps
55a8ec62 Merge pull request #3184 from StoDevX/rn-0.57.5-remove-rc-from-hsp
2e109eeb Merge pull request #3183 from StoDevX/rn-0.57.5--master-sync
7dbfc225 Merge pull request #3180 from StoDevX/jsc-android--236355.0.0
b3db5d0e Merge pull request #3179 from StoDevX/fastlane-refactor
647381c7 Merge pull request #3178 from StoDevX/danger-check-changelog
4f502fb9 Merge pull request #3177 from StoDevX/sync-changelog
46dc636a Merge pull request #3156 from StoDevX/sarn-website
a1aeefd8 Merge pull request #3176 from StoDevX/hours/pause-kitchen
8a22ebe1 Merge pull request #3174 from StoDevX/travis-xenial
d03bfd3f Merge pull request #3166 from StoDevX/remove-credentials-from-redux
d62c38ad Merge pull request #3172 from StoDevX/bundle-update
ec89f24d Merge pull request #3171 from StoDevX/circle-node-10
04be818d Merge pull request #3167 from StoDevX/disable-nav-persistence
6cb26732 Merge pull request #3108 from StoDevX/update-privacy
[...]
```

### CODE REVIEW

**Review by hawkrives — COMMENTED** [review]
We could also be fancy and strip out the common starting text "Merged pull request" - that'd save 19 chars per PR

**Comment by rye** [comment]
@hawkrives, what if we removed `pull request` and `StoDevX/` from each of the merges?

That way we'd still have
```
301ffffa Merge #3196 from renovate-avoid-patterns-where-unneeded
c62b3c48 Merge #3195 from renovate-jsc-android-latest
5ed7f564 Merge #3194 from renovate/pin-dependencies
a940b121 Merge #3193 from renovate/configure
e399b3ec Merge #3045 from greenkeeper/react-native-0.57.0
161270ad Merge #3189 from express-bus-update
40441dab Merge #3185 from rn-0.57.5--remove-hsps
55a8ec62 Merge #3184 from rn-0.57.5-remove-rc-from-hsp
2e109eeb Merge #3183 from rn-0.57.5--master-sync
7dbfc225 Merge #3180 from jsc-android--236355.0.0
b3db5d0e Merge #3179 from fastlane-refactor
647381c7 Merge #3178 from danger-check-changelog
4f502fb9 Merge #3177 from sync-changelog
46dc636a Merge #3156 from sarn-website
a1aeefd8 Merge #3176 from hours/pause-kitchen
8a22ebe1 Merge #3174 from travis-xenial
d03bfd3f Merge #3166 from remove-credentials-from-redux
[...]
```

**Comment by hawkrives** [comment]
What about going even further, and doing like `d03bfd3f - #3166: remove-credentials-from-redux`? (I'm also ok with your suggestion of `d03bfd3f Merge #3166 from remove-credentials-from-redux`.)

**Comment by rye** [comment]
I'd be down for that.

So first we need to do a `.gsub(/pull request |StoDevX\//,'')` and then `.gsub(/Merge/, '—').gsub(' from', ': ')` to achieve the latter.

Those can be chained together.

**Comment by hawkrives** [comment]
@drewvolz was wondering about dropping the commit hash from the lines – so something like 

```
#3196: renovate-avoid-patterns-where-unneeded
#3195: renovate-jsc-android-latest
#3194: renovate/pin-dependencies
#3193: renovate/configure
#3045: greenkeeper/react-native-0.57.0
#3189: express-bus-update
#3185: rn-0.57.5--remove-hsps
#3184: rn-0.57.5-remove-rc-from-hsp
#3183: rn-0.57.5--master-sync
#3180: jsc-android--236355.0.0
#3179: fastlane-refactor
#3178: danger-check-changelog
#3177: sync-changelog
#3156: sarn-website
#3176: hours/pause-kitchen
#3174: travis-xenial
#3166: remove-credentials-from-redux
[...]
```

thoughts? I think I'm OK with it

**Comment by rye** [comment]
Since the commit hash points to the merge commit only, I don't think it's a problem.

This would just involve changing `--oneline` to `--pretty='format:%s'`.

**Comment by hawkrives** [comment]
@rye I can push that

**Comment by rye** [comment]
Nope, I've already got it committed, @hawkrives.

File: fastlane/lib/git.rb [hunk 1]
```diff
@@ -9,14 +9,13 @@ def git_changelog
...
-  # to keep the character count down
-  graph
-    .lines
-    .map(&:chomp)
-    .join "\n"
+  pr_merges = sh("git log #{from_ref}..#{to_ref} --pretty='format:%s' --grep 'Merge pull request'")
+
+  "Merged Pull Requests as of #{describe}:\n\n#{pr_merges}"
+    .gsub(/Merge pull request |StoDevX\//,'')
+    .gsub(' from', ': ')
```
Comment (hawkrives) [inline fastlane/lib/git.rb]: Won't the current version wind up with two spaces after the colon? I don't think we need to print the space after the colon.

```suggestion
    .gsub(' from', ':')
```

**Review by hawkrives — DISMISSED** [review]

File: fastlane/lib/git.rb (hunk 1 above)
Comment (rye) [reply fastlane/lib/git.rb]: Why not replace ` from ` with `: `?  That's clearer to me.  I'll commit that.

Comment (rye) [reply fastlane/lib/git.rb]: Why did GitHub eat my spaces? We'll never know.  I meant to have a space both before and after `from`.

**Review by hawkrives — APPROVED** [review]

**Pull request merged**